src/test/java/core/basesyntax/RegistrationTests.java

```diff
@@ -0,0 +1,126 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationTests {
+    private static User user;
+    private static RegistrationService registrationService;
+    private static StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User(111L, "userLogin", "Rn963b", 18);
+        storageDao.clear();
+    }
+
+    @AfterEach
+    void tearDown() {
+        storageDao.clear();
+    }
+
+    @Test
+    void register_UserWithValidData_OK() {
+        user.setLogin("newLogin333");
+        User actual = registrationService.register(user);
+        User expected = user;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_UserWithAgeLessThanMinimum_NotOK() {
+        user.setAge(16);
```

Comment:
```suggestion
        user.setAge(17);
```
потрібно перевіряти крайні значення